frontend/components/dashboard/new-session.tsx

```diff
@@ -83,10 +138,51 @@ export const NewSession = () => {
                     </SelectContent>
                 </Select>
             </div>
-
             <Button className="mt-4 bg-purple-600 hover:bg-[#A78BFA]" onClick={handleMatchmaking}>
                 Start matchmaking
             </Button>
+
+            {modalData.isOpen && (
+                <CustomModal className={`h-auto w-2/5 rounded-lg items-center`}>
+                    <div className="flex flex-col items-center justify-center space-y-4">
+                        <Image
+                            priority={true}
+                            src="/matchmaking.png"
+                            width={343}
+                            height={234}
+                            alt="pros sitting around a monitor"
+                        />
+                        <h2 className="text-xl font-bold">
+                            {modalData.isMatchmaking ? 'Finding collaborator' : 'Match found!'}
+                        </h2>
+                        {modalData.isMatchmaking && <Loading />}
+                        {modalData.isMatchmaking && (
+                            <h2 className="text-medium font-medium">
+                                {`${Math.floor(timeElapsed / 60)}:${(timeElapsed % 60).toString().padStart(2, '0')}`}
+                            </h2>
+                        )}
+                        {modalData.isMatchFound ? (
+                            <Button
+                                className="mt-4 bg-purple-600 hover:bg-[#A78BFA]"
+                                variant={'primary'}
+                                size={'lg'}
+                                onClick={handleCancelMatchmaking}
+                            >
+                                Proceed to start coding
```

Comment:
Instead of a button I think let's do a redirect, and immediately route to the /code route on successful matchup. There should also be a UI for a timeout match.